fe/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java

```diff
@@ -0,0 +1,114 @@
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.load.routineload;
+
+import com.google.common.base.Strings;
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.common.SystemIdGenerator;
+import org.apache.doris.system.SystemInfoService;
+import org.apache.doris.thrift.TResourceInfo;
+import org.apache.doris.thrift.TTaskType;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+public class KafkaRoutineLoadJob extends RoutineLoadJob {
+    private static final Logger LOG = LogManager.getLogger(KafkaRoutineLoadJob.class);
+
+    private static final String FE_GROUP_ID = "fe_fetch_partitions";
+    private static final int FETCH_PARTITIONS_TIMEOUT = 10;
+
+    private String serverAddress;
+    private String topic;
+    // optional
+    private List<Integer> kafkaPartitions;
+
+    public KafkaRoutineLoadJob() {
+    }
+
+    public KafkaRoutineLoadJob(long id, String name, String userName, long dbId, long tableId,
+                               String partitions, String columns, String where, String columnSeparator,
+                               int desireTaskConcurrentNum, JobState state, DataSourceType dataSourceType,
+                               int maxErrorNum, TResourceInfo resourceInfo, String serverAddress, String topic) {
+        super(id, name, userName, dbId, tableId, partitions, columns, where,
+                columnSeparator, desireTaskConcurrentNum, state, dataSourceType, maxErrorNum, resourceInfo);
+        this.serverAddress = serverAddress;
+        this.topic = topic;
+    }
+
+    @Override
+    public List<RoutineLoadTask> divideRoutineLoadJob(int currentConcurrentTaskNum) {
+        // divide kafkaPartitions into tasks
+        List<KafkaRoutineLoadTask> kafkaRoutineLoadTaskList = new ArrayList<>();
+        for (int i = 0; i < currentConcurrentTaskNum; i++) {
+            // TODO(ml): init load task
+            kafkaRoutineLoadTaskList.add(new KafkaRoutineLoadTask(getResourceInfo(), 0L, TTaskType.PUSH,
+                    dbId, tableId, 0L, 0L, 0L, SystemIdGenerator.getNextId()));
+        }
+        for (int i = 0; i < kafkaPartitions.size(); i++) {
+            kafkaRoutineLoadTaskList.get(i % currentConcurrentTaskNum).addKafkaPartition(kafkaPartitions.get(i));
+        }
+        List<RoutineLoadTask> result = new ArrayList<>();
+        result.addAll(kafkaRoutineLoadTaskList);
+        return result;
+    }
+
+    @Override
+    public int calculateCurrentConcurrentTaskNum() {
+        updatePartitions();
+        SystemInfoService clusterInfo = Catalog.getCurrentSystemInfo();
+        String clusterName = Catalog.getInstance().getDb(dbId).getClusterName();
```

Comment:
1. Use Catalog.getCurrentCatalog(), not getInstance();
2. getDb(dbId) may return null if user drop the database concurrently, you should check it.